src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -300,4 +300,32 @@ export default class ReimbursementRequestService {
 
     return expense;
   }
+
+  /**
+   * Service function to make a reimbursement request as delivered
+   * @param submitter is the User marking the request as delivered
+   * @param requestId is the ID of the reimbursement request to be marked as delivered
+   * @throws NotFoundException if the id is invalid or not there
+   * @throws AccessDeniedException if the creator of the request is not the submitter
+   * @returns the updated reimbursement request
+   */
+  static async markReimbursementRequestAsDelivered(submitter: User, reimbursementRequestId: string) {
+    const reimbursementRequest = await prisma.reimbursement_Request.findUnique({
+      where: { reimbursementRequestId }
+    });
+
+    if (!reimbursementRequest) throw new NotFoundException('Reimbursement Request', reimbursementRequestId);
+
+    if (!(submitter.userId === reimbursementRequest.recipientId))
+      throw new AccessDeniedException('Only the creator of the reimbursement request can mark as delivered');
+
```

Comment:
the more im thinking about it we should just make it so that you can only mark as delivered once, so lets add a check here that if the dateDelivered is not null throw an error saying that you cannot mark an already delivered request